lib/src/main/java/dev/oneuiproject/oneui/layout/ToolbarLayout.java

```diff
@@ -546,6 +551,17 @@ private void setMenuItemBadgeText(ViewGroup mBadgeBackground, TextView mBadgeTex
         ViewGroup.MarginLayoutParams lp = (MarginLayoutParams) mBadgeBackground.getLayoutParams();
         lp.setMarginEnd(0);
         lp.width = (int) (getResources().getDimension(androidx.appcompat.R.dimen.sesl_badge_default_width) + (text.length() * getResources().getDimension(androidx.appcompat.R.dimen.sesl_badge_additional_width)));
+        lp.height = (int) (getResources().getDimension(androidx.appcompat.R.dimen.sesl_badge_default_width) + (text.length() * getResources().getDimension(androidx.appcompat.R.dimen.sesl_badge_additional_width)));
+        mBadgeBackground.setLayoutParams(lp);
+    }
+
+    private void setMenuItemBadgeDot(ViewGroup mBadgeBackground, Boolean status) {
+        mBadgeBackground.setVisibility(status == null || !status ? GONE : VISIBLE);
+        if (status == null) return;
+        ViewGroup.MarginLayoutParams lp = (MarginLayoutParams) mBadgeBackground.getLayoutParams();
+        lp.setMarginEnd(0);
+        lp.width = (int) (getResources().getDimension(androidx.appcompat.R.dimen.sesl_menu_item_badge_size));
+        lp.height = (int) (getResources().getDimension(androidx.appcompat.R.dimen.sesl_menu_item_badge_size));
```

Comment:
For efficiency, retrieve value of same resource only once and use its cached value on other places. Also cache getResources().

Reply:
Done!! check again!